pkg/task/serialize_test.go

```diff
@@ -95,11 +105,76 @@ func TestSerializeHttpPayload(t *testing.T) {
 			},
 			wantErr: false,
 		},
+		{
+			name:        "list of strings",
+			body:        `{}`,
+			queryParams: url.Values{"listOfStrings": []string{"a", "b", "c"}},
+			wantPayload: &types.TaskPayload{
+				Args:   nil,
+				Kwargs: map[string]interface{}{"listOfStrings": []string{"a", "b", "c"}},
+			},
+			wantErr: false,
+		},
+		{
+			name:        "single number",
+			body:        `{}`,
+			queryParams: url.Values{"sleep": []string{"100"}},
+			wantPayload: &types.TaskPayload{
+				Args:   nil,
+				Kwargs: map[string]interface{}{"sleep": float64(100)},
+			},
+			wantErr: false,
+		},
+		{
+			name:        "single number no body",
+			queryParams: url.Values{"sleep": []string{"100"}},
+			wantPayload: &types.TaskPayload{
+				Args:   nil,
+				Kwargs: map[string]interface{}{"sleep": float64(100)},
+			},
+			wantErr: false,
+		},
+		{
+			name:        "list of ints",
+			queryParams: url.Values{"sleep": []string{"100", "200", "300"}},
+			wantPayload: &types.TaskPayload{
+				Args:   nil,
+				Kwargs: map[string]interface{}{"sleep": []float64{100, 200, 300}},
+			},
+			wantErr: false,
+		},
+		{
+			name:        "list of floats",
+			queryParams: url.Values{"sleep": []string{"100.1", "200.2", "300.3"}},
+			wantPayload: &types.TaskPayload{
+				Args:   nil,
+				Kwargs: map[string]interface{}{"sleep": []float64{100.1, 200.2, 300.3}},
+			},
+			wantErr: false,
+		},
+		{
+			name:        "list of mixed ints and floats",
+			queryParams: url.Values{"sleep": []string{"100", "200.2", "300"}},
+			wantPayload: &types.TaskPayload{
+				Args:   nil,
+				Kwargs: map[string]interface{}{"sleep": []float64{100.0, 200.2, 300.0}},
+			},
+			wantErr: false,
+		},
+		{
+			name:        "mix of strings and numbers",
+			queryParams: url.Values{"sleep": []string{"Today", "200.2", "300"}},
+			wantPayload: &types.TaskPayload{
+				Args:   nil,
+				Kwargs: map[string]interface{}{"sleep": []string{"Today", "200.2", "300"}},
+			},
+			wantErr: false,
+		},
```

Comment:
With fastapi, you can define an endpoint like below to take a list of mixed types.  
```python
@app.get("/endpoint")
async def endpoint(stuff: List[Union[int, str]] = Query(None)):
    return {"stuff": stuff}
```

With our endpoints, that will not work currently. Everything will become a string. We could try to create a new type that we then unravel into the correct types in the runner. Alternatively, we can just document this divergence and leave it as is.